src/trainer_card.c

```diff
@@ -0,0 +1,311 @@
+/*static*/ void sub_80C26D4(void);
+/*static*/ void sub_80C48C8(void);
+/*static*/ void sub_80C2710(void);
+/*static*/ void sub_80C2728(u8 task);
+/*static*/ bool8 sub_80C3438(void);
+/*static*/ void sub_80C438C(u8);
+/*static*/ void sub_80C4FF0(void);
+/*static*/ void sub_80C4550(u8*); 
+/*static*/ void sub_80C45C0(u8*);
+/*static*/ void sub_80C4630(void);
+/*static*/ void sub_80C3880(void);
+/*static*/ void sub_80C4918(void);
+/*static*/ bool8 sub_80C4940(void);
+/*static*/ bool8 sub_80C2AD8(void);
+/*static*/ void sub_80C2C80(void);
+
+extern struct UnknownStruct{
+    u8 var_0;
+    u8 var_1;
+    u8 var_2;
+    u8 filler3[2];
+    u8 var_5;
+    u8 filler6[3];
+    u8 var_9;
+    u8 fillerA[0x51F];
+    u8 var_529;
+    u8 var_52A;
+    u8 var_52B;
+    u16 var_52C;
+    void (*callback2)(void);
+    u8 filler531[0x64];
+    u8 var_598[0x4B0];
+    u8 var_A48[0x4B0];
+    u8 var_EF8[0x4B0];
+    u8 var_13A8[0x400];
+    u8 var_17A8[0x200];
+    u8 var_19A8[0x200];
+}* gUnknown_02039CE8;
+
+void sub_80C2690(void)
+{
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+    sub_80C48C8();
+    if(gUnknown_02039CE8->var_9)
+        DmaCopy16(3, &gScanlineEffectRegBuffers[0], &gScanlineEffectRegBuffers[1], 0x140);
+}
+
+void sub_80C26D4(void)
+{
+    u16 backup;
+    u16 bgVOffset;
+    
+    backup = REG_IME;
+    REG_IME = 0;
+    bgVOffset = gScanlineEffectRegBuffers[1][REG_VCOUNT & 0xFF];
+    REG_BG0VOFS = bgVOffset;
+    REG_IME = backup;
+}
+
+void sub_80C2710(void)
+{
+    RunTasks();
+	AnimateSprites();
+	BuildOamBuffer();
+	UpdatePaletteFade();
+}
+
+void sub_80C2728(u8 taskId)
+{
+    SetMainCallback2(gUnknown_02039CE8->callback2);
+    FreeAllWindowBuffers();
+    Free(gUnknown_02039CE8);
+    gUnknown_02039CE8 = NULL;
+    DestroyTask(taskId);
+} 
+
+void sub_80C2760(u8 taskId)
+{
+    switch(gUnknown_02039CE8->var_0)
+    {
+    case 0:    
+        if(!IsDma3ManagerBusyWithBgCopy())
+        {
+            FillWindowPixelBuffer(1, 0);
+            gUnknown_02039CE8->var_0++;
+        }
+        break;
+    case 1:
+        if(sub_80C3438())
+            gUnknown_02039CE8->var_0++;
+        break;
+    case 2:
+        sub_80C438C(1);
+        gUnknown_02039CE8->var_0++;
+        break;
+    case 3:
+        FillWindowPixelBuffer(2, 0);
+        sub_80C4FF0();
+        sub_80C438C(2);
+        gUnknown_02039CE8->var_0++;
+        break;
+    case 4:
+        sub_80C4550(gUnknown_02039CE8->var_EF8);
+        gUnknown_02039CE8->var_0++;
+        break;
+    case 5:
+        sub_80C45C0(gUnknown_02039CE8->var_598);
+        gUnknown_02039CE8->var_0++;
+        break;
+    case 6:
+        sub_80C4630();
+        gUnknown_02039CE8->var_0++;
+        break;
+    case 7:
+        if(gWirelessCommType == TRUE && gReceivedRemoteLinkPlayers == TRUE)
+        {
+            sub_800E0E8();
+            CreateWirelessStatusIndicatorSprite(230, 150);
+        }
+        BlendPalettes(-1, 16, gUnknown_02039CE8->var_52C);
```

Comment:
palette blending functions take a bitmask as their first argument to select which palettes it's applying to.  Therefore, use `0xFFFFFFFF` instead of  -1